src/main/java/org/teragrep/JavaPcre.java

```diff
@@ -147,8 +216,12 @@ public void pcre2_singlematch_java(String a, int b){
         int ind = 0;
 
 
-        LibJavaPcre.RegexStruct.ByValue regex_val = LibJavaPcre.INSTANCE.pcre2_single_jmatch(subject, re, offset, match_options);
-        if (regex_val.numVals == 0) {
+        LibJavaPcre.RegexStruct.ByValue regex_val = LibJavaPcre.INSTANCE.pcre2_single_jmatch(subject, re, offset, match_options, mcontext);
+        if (regex_val.rc < 0) {
+            switch(regex_val.rc){
+                case -1: LOGGER.warn("No match"); break; // rc = -1 should be equal to rc = PCRE2_ERROR_NOMATCH in C.
+                default: LOGGER.error("Matching error:" + regex_val.rc); break; // anything lower than -1 is a matching error.
```

Comment:
do we know all the possibile values of rc? It may be preferrable to create an exception and sub-class it in case the exceptions can be handled by the calling code.

Creating a specific exception class example can be found here:
https://github.com/teragrep/rlo_06/blob/master/src/main/java/com/teragrep/rlo_06/ParseException.java#L48

I'd recommend to create a PcreException and further sub-class it to the specific types.

Reply:
All possible matching errors are indeed documented, but it seems that compile errors are not documented at all and instead pcre2_get_error_message() function should be used within the library for translating the error code to textual description of the error. Changed both matching and compiling to use that function for exception handling.